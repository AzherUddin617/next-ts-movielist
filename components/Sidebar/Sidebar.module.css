.sidebar {
  @apply px-6 pb-6;
}

.sidebar .container {
  @apply flex md:flex-col items-center h-full p-4 bg-secondary-dark;
  @apply rounded-md shadow-sm;
}

.sidebar .container .logo {
  @apply pr-6 md:pr-0 md:pb-6 text-primary text-3xl;
}
.sidebar .container .logo .icon {
  @appl y cursor-pointer;
}

.sidebar .container .list {
  @apply flex-1 flex md:flex-col items-center text-secondary text-xl;
}
.sidebar .container .list .item {
  @apply px-2 md:px-0 py-0 md:py-2 cursor-pointer transition-colors;
  /* @apply opacity-75 hover:opacity-100 active:text-white; */
  @apply hover:text-secondary-black;
}
.sidebar .container .list .item .icon {
}

.sidebar .container .user {
  @apply text-3xl opacity-75 hover:opacity-100 transition-opacity cursor-pointer;
}
