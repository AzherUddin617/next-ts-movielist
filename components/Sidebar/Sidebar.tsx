import Link from 'next/link'

import classes from './Sidebar.module.css'
import { MdMovie } from 'react-icons/md'
import { FaThList, FaBookmark, FaRegBookmark, FaRegUserCircle } from 'react-icons/fa'
import { HiTrendingUp } from 'react-icons/hi'

const contents = [
    { title: 'Home', Icon: FaThList, link: '/' },
    { title: 'Top Rated', Icon: HiTrendingUp, link: '/top' },
    { title: 'Bookmark', Icon: FaBookmark, link: '/bookmark' },
]

const Sidebar = () => {
  return (
    <div className={classes.sidebar}>
        <div className={classes.container}>

            <div className={classes.logo}>
                <Link href='/'>
                    <MdMovie className={classes.icon} />
                </Link>
            </div>

            <ul className={classes.list}>
                {contents.map((content, index) => (
                    <Link key={index} href={content.link}>
                        <li className={classes.item}>
                            <content.Icon className={classes.icon} />
                        </li>
                    </Link>
                ))}
            </ul>

            <div className={classes.user}>
                <FaRegUserCircle className={classes.icon} />
            </div>

        </div>
    </div>
  )
}

export default Sidebar