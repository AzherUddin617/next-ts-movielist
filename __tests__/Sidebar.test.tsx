/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from '@testing-library/react'
// import Home from '../pages/index'
import Sidebar from '@/components/Sidebar/Sidebar';

describe('Sidebar', () => {
  it('renders list', () => {
    render(<Sidebar />);
    expect(screen.getByRole('list')).toBeInTheDocument();
  })

  it("has list item with Home title", () => {
    const { getByTitle } = render(<Sidebar />);
    expect(getByTitle(/home/i)).toBeInTheDocument();
  })
  
  it("has list item with Bookmark title", () => {
    const { getByTitle, container } = render(<Sidebar />);
    expect(getByTitle(/bookmark/i)).toBeInTheDocument();
    expect(container.querySelector('.icon')).toBeVisible();
  })

  it("has list item with Bookmark title", () => {
    const { container } = render(<Sidebar />);
    expect(container.querySelector('.icon')).toBeVisible();
  })

  it("has 3 listitem with title in the list", () => {
    const { container } = render(<Sidebar />);
    expect(container.querySelectorAll('.item').length).toBe(3);

    const listItem = screen.getAllByRole('listitem')
    expect(listItem.length).toBe(3)
    listItem.forEach(item => {
      expect(item).toHaveProperty('title')
    })
  })
})

describe("Sidebar snapshots", ()=> {
  it ("renders correctly", ()=> {
    const tree = render(<Sidebar />)
    expect(tree).toMatchSnapshot();
  })
})