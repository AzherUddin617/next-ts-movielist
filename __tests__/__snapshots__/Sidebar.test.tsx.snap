// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar snapshots renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebar"
      >
        <div
          class="container"
        >
          <div
            class="logo"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
              />
            </svg>
          </div>
          <ul
            class="list"
          >
            <li
              class="item"
              title="Home"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M149.333 216v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-80c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zM125.333 32H24C10.745 32 0 42.745 0 56v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24zm80 448H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm-24-424v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24zm24 264H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24z"
                />
              </svg>
            </li>
            <li
              class="item"
              title="Top Rated"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12 7a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0V8.414l-4.293 4.293a1 1 0 01-1.414 0L8 10.414l-4.293 4.293a1 1 0 01-1.414-1.414l5-5a1 1 0 011.414 0L11 10.586 14.586 7H12z"
                  fill-rule="evenodd"
                />
              </svg>
            </li>
            <li
              class="item"
              title="Bookmark"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
                />
              </svg>
            </li>
          </ul>
          <div
            class="user"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebar"
    >
      <div
        class="container"
      >
        <div
          class="logo"
        >
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
            />
          </svg>
        </div>
        <ul
          class="list"
        >
          <li
            class="item"
            title="Home"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M149.333 216v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-80c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zM125.333 32H24C10.745 32 0 42.745 0 56v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24zm80 448H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm-24-424v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24zm24 264H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24z"
              />
            </svg>
          </li>
          <li
            class="item"
            title="Top Rated"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12 7a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0V8.414l-4.293 4.293a1 1 0 01-1.414 0L8 10.414l-4.293 4.293a1 1 0 01-1.414-1.414l5-5a1 1 0 011.414 0L11 10.586 14.586 7H12z"
                fill-rule="evenodd"
              />
            </svg>
          </li>
          <li
            class="item"
            title="Bookmark"
          >
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
              />
            </svg>
          </li>
        </ul>
        <div
          class="user"
        >
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
